packages/compat/src/resolver-transform.ts

```diff
@@ -329,15 +329,12 @@ function handleComponentHelper(
   resolver.resolveComponentHelper(locator, moduleName, param.loc, impliedBecause);
 }
 
-function handleDynamicHelper(param: ASTv1.Node, resolver: Resolver, moduleName: string): void {
-  switch (param.type) {
-    case 'StringLiteral':
-      resolver.resolveDynamicHelper({ type: 'literal', path: param.value }, moduleName, param.loc);
-      break;
-    case 'TextNode':
-      resolver.resolveDynamicHelper({ type: 'literal', path: param.chars }, moduleName, param.loc);
-      break;
-    default:
-      resolver.resolveDynamicHelper({ type: 'other' }, moduleName, param.loc);
+function handleDynamicHelper(param: ASTv1.Expression, resolver: Resolver, moduleName: string): void {
```

Comment:
I'm new to TypeScript, but I think `Expression` is the correct type here since `node.params` seems to be typed as `Expression[]`? `Expression` doesn't include the `TextNode` type, so that doesn't need to be covered anymore.